daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/layers/api/Layer.java

```diff
@@ -96,23 +275,34 @@ public Layer(String localName, String targetNamespace) {
     this.targetNamespace = targetNamespace;
   }
 
-  /** The spiName of the Layer class.
+  /** For framework use.
    * <p>
    * This method and the string it returns are required by the SPI loader.
    * @return A unique indentifier for the kind of layer. Contains both local and namespace components of the layer's complete name.
    */
   public final String name() { return LayerUtils.spiName(localName, targetNamespace); }
 
+  /** For framework use. */
   public final String localName() { return this.localName; }
+
+  /** For framework use. */
```

Comment:
We might want to consider an update (in a later release) to make these package private, which I think should exclude them from javadoc? I guess that would mean the classes calling these functions would need to be in the same package--maybe that refactoring would be too complicated.

Reply:
Yeah, package private requires use of reflection, or lots of other code has to be in the layers.api package. 

However, ChatGPT4o says there are ways to add one's own annotations to java code that the javadoc tool can see:

https://chatgpt.com/share/ed29c152-9bfa-4598-8a46-bf590abefb7d

Perhaps in the future. 

Note that if you are pointing an IDE at the jar and sources, this wouldn't work. only the generated javadoc HTML would see the exclusions.